client/ayon_applications/plugins/launcher_actions/debug_shell.py

```diff
@@ -0,0 +1,116 @@
+from typing import Optional
+import os
+import subprocess
+
+from qtpy import QtWidgets, QtGui, QtCore
+
+from ayon_applications import (
+    Application,
+    ApplicationManager,
+)
+from ayon_applications.utils import get_app_environments_for_context
+from ayon_core.pipeline import LauncherAction
+from ayon_core.style import load_stylesheet
+
+
+def get_application_qt_icon(
+        application: Application
+) -> Optional[QtGui.QIcon]:
+    """Return QtGui.QIcon for an Application"""
+    # TODO: Improve workflow to get the icon, remove 'color' hack
+    from ayon_core.tools.launcher.models.actions import get_action_icon
+    from ayon_core.tools.utils.lib import get_qt_icon
+    application.color = "white"
+    return get_qt_icon(get_action_icon(application))
+
+
+class DebugShell(LauncherAction):
+    """Run any host environment in command line."""
+    name = "debugshell"
+    label = "Shell"
+    icon = "terminal"
+    color = "#e8770e"
+    order = 10
+
+    def is_compatible(self, selection) -> bool:
+        return all((
+            selection.is_task_selected,
+            selection.is_folder_selected,
+            selection.is_project_selected,
+        ))
```

Comment:
You can't select task without having selected folder or project.
```suggestion
        return selection.is_task_selected
```